src/service/url-replacements-impl.js

```diff
@@ -540,6 +547,17 @@ export class UrlReplacements {
     return navigationInfo[attribute];
   }
 
+  getQueryParamData_(param, defaultValue, opt_sync) {
+    user().assert(param,
+        'The first argument to QUERY_PARAM, the query string ' +
+        'param is required');
+    const url = parseUrl(this.ampdoc.win.location.href);
```

Comment:
This doesn't have the same problem as above, because it reads the `.href` every time.